addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -34,6 +34,18 @@ def _compute_display_action_record_components(self):
             if subcontracted_moves._subcontrating_can_be_record():
                 picking.display_action_record_components = 'facultative'
 
+    @api.depends('picking_type_id', 'partner_id')
+    def _compute_location_id(self):
+        super(StockPicking, self)._compute_location_id()
+
+        for picking in self:
+            # If this is a subcontractor resupply transfer, set the destination location
+            # to the vendor subcontractor location
+            if picking.picking_type_id.default_location_dest_id.is_subcontracting_location\
```

Comment:
I would check the `picking_type_id` is the one set in the `subcontracting_resupply_type_id` of the warehouse instead. It's stronger than checking the location